Add Copilot Coding Agent integration post and related assets (#8608)
Microsoft/vscode-docs: blogs/2025/07/16/copilot-coding-agent.md

## What is GitHub Copilot Coding Agent?

[GitHub Copilot Coding Agent](https://github.blog/changelog/2025-07-14-start-and-track-github-copilot-coding-agent-sessions-from-visual-studio-code/) is an autonomous AI developer you can assign to any GitHub issue. The agent runs in GitHub and is fully integrated as a member of your repo that can comment on issues, open PRs, do code reviews and more. The agent can tackle complex work, implement features across your codebase, and even use Model Context Protocol (MCP) tools to work with databases or cloud services.

1. You [enable the agent in your settings](https://docs.github.com/en/copilot/how-tos/agents/copilot-coding-agent/enabling-copilot-coding-agent).
2. You assign an issue to Copilot.
3. The agent does the work, opens a pull request, and keeps you in the loop.
4. You review the PR, leave comments if you want changes, and the agent iterates until it’s done.

![Copilot Coding Agent on GitHub](coding-agent-github.png)

That's all there is to it. It's the workflow you already know with an agent in the mix.

> The agent runs within a temporary, isolated dev environment that gets spun up where the agent can explore the codebase, make changes, build the code, run tests, etc. - a complete dev enviornment just for the agent so it can function in a fully autonomous manner.

Reviewer: ```suggestion
The agent runs within a temporary, isolated dev environment that gets spun up where the agent can explore the codebase, make changes, build the code, run tests, etc. - a complete dev environment just for the agent so it can function in a fully autonomous manner.
```

![Live preview of the site on Netlify but we have an error](live-preview-netlify.png)

Uh oh - now that we look closely, it looks like we didn't quite get the "Trending" section working. In this case, that's ok. In fact it's good. Our agent couldn't succeed in one shot because it doesn't have access to create the required database changes. Instead, it created a migration script as part of the PR. That *will* require us to manually check out the PR, but you might feel more comfortable being in control over that sort of change.

But what if we *did* want to give access to make the database changes to a dev environment? Coding Agent has support for MCP servers, and provided you have one for your database of choice (in this case Supabase), you can [configure MCP servers for the agent on the repo](https://docs.github.com/en/copilot/how-tos/agents/copilot-coding-agent/extending-copilot-coding-agent-with-mcp). It's up to you - how autonomous do you want the agent to be?

![MCP configuration shown for Copilot Coding Agent on github.com](coding-agent-mcp-config.png)

But it gets even better - because you can do all of this right from VS Code.

### Start sessions from Copilot Chat

> We find ourselves wanting to do more and more right from the [chat panel](https://code.visualstudio.com/docs/copilot/chat/copilot-chat) in VS Code. This is where most work and thinking happens and you're not locked into the issue workflow. You can - at any point in a chat, no matter how long - delegate the task directly to the Coding Agent. All of the context that is currently a part of that chat will get sent to the Coding Agent.

Reviewer 2: made some minor tweaks to phrasing but implemented these suggestions!

The Coding Agent then opens a PR and goes directly to work - skipping the issue step entirely. It distills down the context into a detailed PR description and you'll see that Copilot Coding Agent makes use of "to-do lists" so that you can see what it's doing and where it is in the process at a glance.

<video src="delegate-to-coding-agent.mp4" title="Delegate to Coding Agent video" controls poster="delegate-to-coding-agent.png"></video>

## What’s next?

The Copilot Coding Agent integration in Visual Studio Code is currently in preview, but you can use it all today. If you want to light all of this up in your editor, all you need to do is make sure you have the GitHub Pull Requests extension installed and add the following setting...

```json
"githubPullRequests.codingAgent.uiIntegration": true,

> ```

Reviewer 3: I just realized this is at the very end - but a lot of people won't make it this far. We gotta make this the "call to action" and bring it to the very top. The flow is

1. We've got a sneak peek of something big - but you can use it today.
2. Install this extension and add this one setting, and check out what you get
3. \[Rest of blog post]

Reviewer 3: We can repeat this at the end in case people missed it - "as a reminder, this is in preview, and you'll need to enable `"githubPullRequests.codingAgent.uiIntegration": true,`

Reviewer 2: added a mention at the top, lmk what you think

We are very excited to further deepen this integration with Coding Agent. And since we recently made the move [to fully open source the AI features in VS Code](https://code.visualstudio.com/blogs/2025/05/19/openSourceAIEditor), you can track our progress in our [Coding Agent iteration plan](https://github.com/microsoft/vscode/issues/255483). We're working on:
- PR performance and rendering
- Integrated chat view for Coding Agent sessions
- Bringing the Copilot Agents command center into VS Code
- Sharing custom instructions between the Coding Agent and VS Code
- Adding more documentation

We'd love your thoughts on the experience, so please provide any feedback as [issues in the VS Code repo](https://github.com/microsoft/vscode/issues)!

## Go forth and 100x yourself

We can’t wait to see what you and your new agent friends build together! Keep an eye on the [release notes](/release-notes/) and the [Copilot documentation](https://docs.github.com/en/copilot/concepts/about-copilot-coding-agent) for new developments here.

> In the meantime, start making a list of all the things you’d rather not do. Because you’re about to have a new favorite teammate.

Reviewer 3: Nit: we talk about a new favorite teammate, but the intro is about having a crew. 😄

Reviewer 2: updated to keep the squad of teammates reference!

To see this in action in your editor right now, install the [GitHub Pull Requests extension](https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-pull-request-github) and add the following setting...

```json
"githubPullRequests.codingAgent.uiIntegration": true,
```

In this post, we'll take a look at what the Coding Agent is, show you an exclusive preview of the integrations in VS Code, and give you a peak at what's coming next.

## What is GitHub Copilot Coding Agent?

[GitHub Copilot Coding Agent](https://github.blog/changelog/2025-07-14-start-and-track-github-copilot-coding-agent-sessions-from-visual-studio-code/) is an autonomous AI developer you can assign to any GitHub issue. The agent runs in GitHub and is fully integrated as a member of your repo that can comment on issues, open PRs, do code reviews, and more. The agent can tackle complex work, implement features across your codebase, and even use Model Context Protocol (MCP) tools to work with databases or cloud services.

It works like this:

> 1. You [enable the agent in your settings](https://docs.github.com/en/copilot/how-tos/agents/copilot-coding-agent/enabling-copilot-coding-agent).

Reviewer 3: ```suggestion
1. You [enable the agent in your GitHub settings](https://docs.github.com/en/copilot/how-tos/agents/copilot-coding-agent/enabling-copilot-coding-agent).
```

2. You assign an issue to Copilot.
3. The agent does the work, opens a pull request, and keeps you in the loop.
4. You review the PR, leave comments if you want changes, and the agent iterates until it’s done.

![Copilot Coding Agent on GitHub](coding-agent-github.png)

That's all there is to it. It's the workflow you already know with an agent in the mix.

The agent runs within a temporary isolated dev environment that gets spun up where the agent can explore the codebase, make changes, build the code, run tests, etc. - a complete dev environment just for the agent so it can function in a fully autonomous manner.

> And now, we're integrating the Coding Agent and all of its workflows directly into VS Code!

Reviewer 2: i don't think this is technically right - pulled from the github blog "Copilot coding agent is also integrated into github.com, GitHub Mobile, the GitHub CLI, and the GitHub MCP Server. "

## Copilot Coding Agent in Visual Studio Code

Copilot Coding Agent integrates with the [GitHub Pull Requests extension](https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-pull-request-github). You can assign any issue to Copilot Coding Agent right from the sidebar - no need to switch to your browser. Just select, assign, and let the agent take it from there.

![A GIF showing Copilot assigned to an issue from within VS Code](assign-to-copilot-gif.gif)

> As with every VS Code feature, the VS Code team itself uses the Coding Agent integration in their own development. In fact, you can see just how many PRs have been [assigned to Copilot Coding Agent in the VS Code repo](https://github.com/search?q=is%3Apr+head%3Acopilot%2F+repo%3Amicrosoft%2Fvscode&type=pullrequests), and that number has continued to increase.

Reviewer 3: Maybe I was a little wrong on this one:
```suggestion
As with every VS Code feature, the VS Code team itself uses the Coding Agent integration in their own development. In fact, you can see just how many PRs have been [assigned to Copilot Coding Agent in the VS Code repo](https://github.com/search?q=is%3Apr+head%3Acopilot%2F+repo%3Amicrosoft%2Fvscode&type=pullrequests), and you can watch that number continue to increase!
```

But it gets even better - because you can do all of this right from VS Code.

### Start sessions from Copilot Chat

We find ourselves wanting to do more and more right from the [chat panel](https://code.visualstudio.com/docs/copilot/chat/copilot-chat) in VS Code. While we love working in GitHub, many of us find that VS Code is our home base. Bringing Coding Agent into chat means you're not locked into the issue workflow. Now, you can - at any point in a chat, no matter how long - delegate the task you're working on directly to the Coding Agent. All of the context that's part of your current chat session will be handed off to the Coding Agent.

The Coding Agent then opens a PR and goes directly to work - skipping the issue step entirely. It distills down the context into a detailed PR description and you'll see that Copilot Coding Agent makes use of "to-do lists" so that you can see what it's doing and where it is in the process at a glance.

<video src="delegate-to-coding-agent.mp4" title="Delegate to Coding Agent video" controls poster="delegate-to-coding-agent.png"></video>

## What’s next?

> The Copilot Coding Agent integration in Visual Studio Code is currently in preview, but you can use it all today. As a reminder, make sure you have the GitHub Pull Requests extension installed and add the following setting...

Reviewer 3: ```suggestion
Keep in mind, the Copilot Coding Agent integration in Visual Studio Code is currently in preview - but you can use it all today! Just make sure you have the [GitHub Pull Requests extension](https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-pull-request-github) installed and add the following setting to VS Code:
```

Reviewer 2: made some slight tweaks but implemented this suggestion!

```json
"githubPullRequests.codingAgent.uiIntegration": true,
```

We are very excited to further deepen this integration with Coding Agent. And since we recently made the move [to fully open source the AI features in VS Code](https://code.visualstudio.com/blogs/2025/05/19/openSourceAIEditor), you can track our progress in our [Coding Agent iteration plan](https://github.com/microsoft/vscode/issues/255483). We're working on:
- PR performance and rendering
- Integrated chat view for Coding Agent sessions
- Bringing the Copilot Agents command center into VS Code
- Sharing custom instructions between the Coding Agent and VS Code
- Adding more documentation

We'd love your thoughts on the experience, so please provide any feedback as [issues in the VS Code repo](https://github.com/microsoft/vscode/issues)!

## Go forth and 100x yourself

We can’t wait to see what you and your new agent friends build together! Keep an eye on the [release notes](/release-notes/) and the [Copilot documentation](https://docs.github.com/en/copilot/concepts/about-copilot-coding-agent) for new developments here.

> In the meantime, start making a list of all the things you’d rather not do. Because you’re about to have a new favorite squad of teammates.

Reviewer 3: This is good - we can also go with

```suggestion
In the meantime, start making a list of all the things you’d rather not do. Because you’re about to have a few new favorite teammates.
```
